### DISCUSSION TITLE
D3 version when importing from CDN

### DISCUSSION THREAD

**Discussion by GerardoFurtado** [d3/d3#3631] [Q&A] [answered]
Importing D3 from jsDelivr...

    import * as d3 from "https://cdn.jsdelivr.net/npm/d3@7/+esm";

...will not expose the version string (`d3.version`), I believe the reason is that the modules are requested separately. 

Is there any alternative for finding out the version numbers of the *equivalent* bundle (that is, without `+esm`), or are the version numbers created only for the bundle? I know that just by looking at `d3@7` we know that the major version is 7, but suppose I want to find out the minor and patch version numbers (without specifying them).

Thanks,

**Comment by Fil** [comment]
and you don't want to parse https://cdn.jsdelivr.net/npm/d3@7/+esm? All the version numbers are in this file, it seems.

**Comment by curran** [comment, marked as answer]
FWIW If you craft this URL, it generates a page you can navigate, which also shows the version number:

https://cdn.jsdelivr.net/npm/d3/

![image](https://user-images.githubusercontent.com/68416/216350577-137f0833-de6b-4780-b36f-b52a73a0913a.png)
